### PR TITLE
applications: nrf5340_audio: Add DFU support.

### PR DESCRIPTION
**Pull request by pirun**
This PR enables [FOTA over Bluetooth LE](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf5340.html#fota-upgrades)

Add config overlay to enable MCUBOOT.
.MCUBOOT swap method need a page(4096bytes) for image metadata, reduce debug build footprint.
Add config and DTC overlay to enable MCUBOOT SECONDARY external which needs External Flash Shield.
Add new buildprog.py parameter.
. -m internal is to enable mcuboot with internal flash.
. -m external is to enable mcuboot with external flash.
. -M is to use minimal size of network core bootloader.
Add static layout for 2MB/8MB NOR external flash.
Signed-off-by: Pirun Lee <pirun.lee@nordicsemi.no>

### CODE REVIEW

File: applications/nrf5340_audio/tools/buildprog/buildprog.py [hunk 1]
```diff
@@ -224,7 +254,33 @@ def __main():
                         const=True,
                         help="Run nrfjprog sequentially instead of in \
                         parallel.")
+    #DFU relative option
+    parser.add_argument("-m", "--min_b0n", default=False,
+                        action='store_true',
+                        help="net core bootloader use minimal size build")
+    group = parser.add_mutually_exclusive_group()
```
Comment (erik-nordic) [inline applications/nrf5340_audio/tools/buildprog/buildprog.py:261]: Maybe it is easier to have a --mcuboot  {external, internal} argument with choices internal flash or external flash?

File: applications/nrf5340_audio/tools/buildprog/buildprog.py [hunk 2]
```diff
@@ -224,7 +254,33 @@ def __main():
...
+    group.add_argument("-M", "--mcuboot", default=False,
+                        action='store_true',
+                        help="enable internal flash dfu")
+    group.add_argument("-e", "--external_flash", default=False,
+                        action='store_true',
+                        help="enable external flash dfu")
+
     options = parser.parse_args(args=sys.argv[1:])
+    global USE_MCUBOOT
```
Comment (erik-nordic) [inline applications/nrf5340_audio/tools/buildprog/buildprog.py:270]: and here it might be nicer to use args.mcuboot instead of global variables. I would also pass these options to the function that uses them, instead of relying on a global parameter, but that is personal preference.

File: applications/nrf5340_audio/tools/buildprog/buildprog.py [hunk 3]
```diff
@@ -224,7 +254,33 @@ def __main():
...
+    global USE_MIN_B0N
+    global PWD
+
+    if options.min_b0n:
+        USE_MIN_B0N = options.min_b0n
+    if options.mcuboot:
+        USE_MCUBOOT = options.mcuboot
+    elif options.external_flash:
+        USE_EXTERNAL_FLASH = options.external_flash
+    PWD=os.getcwd()
```
Comment (erik-nordic) [inline applications/nrf5340_audio/tools/buildprog/buildprog.py:281]: Depending on if https://github.com/nrfconnect/sdk-nrf/pull/7542 get merged first, this will no longer be necessary.

**Review by erik-nordic — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 1 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py:261]: Sure.
Will refactor after https://github.com/nrfconnect/sdk-nrf/pull/7542 merged.

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 3 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py:281]: Sure.
Will refactor after https://github.com/nrfconnect/sdk-nrf/pull/7542 merged.

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 1 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py]: Change parameter to --mcuboot {external, internal} stype.

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 2 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py]: Add options parameter to related functions.

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 3 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py]: Use NRF5340_AUDIO_FOLDER insteal PWD+ relative path.

File: applications/nrf5340_audio/overlay-dfu.conf [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+CONFIG_THREAD_NAME=y
+# DFU
+CONFIG_MCUMGR=y
+CONFIG_PCD_APP=y
+CONFIG_FLASH=y
+CONFIG_MCUMGR_CMD_OS_MGMT=y
+CONFIG_MCUMGR_CMD_IMG_MGMT=y
+CONFIG_MCUMGR_BUF_SIZE=1024
+
+# BT dfu
```
Comment (rick1082) [inline applications/nrf5340_audio/overlay-dfu.conf:25]: Repace "dfu" to "DFU over BLE" would be great.

File: applications/nrf5340_audio/overlay-dfu.conf [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+# Serial DFU
+#CONFIG_MCUMGR_SMP_UART=y
+#CONFIG_UART_MCUMGR_RX_BUF_SIZE=512
+## Required by the `taskstat` command.
+CONFIG_THREAD_MONITOR=y
+CONFIG_STATS=y
+CONFIG_STATS_NAMES=y
+CONFIG_MCUMGR_CMD_STAT_MGMT=y
+# Recude footprint for debug build
```
Comment (rick1082) [inline applications/nrf5340_audio/overlay-dfu.conf:41]: Typo: Reduce

File: applications/nrf5340_audio/overlay-dfu_external_flash.conf [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+# provide a memory region that is used to forward the new firmware to the network core.
+CONFIG_FLASH_SIMULATOR=y
+CONFIG_FLASH_SIMULATOR_DOUBLE_WRITES=y
+CONFIG_FLASH_SIMULATOR_STATS=n
+
+# Increase system workqueue stack size to prevent stack overflows while handling
+# SMP requests.
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=8192
+CONFIG_MAIN_STACK_SIZE=4096
+CONFIG_THREAD_NAME=y
```
Comment (rick1082) [inline applications/nrf5340_audio/overlay-dfu_external_flash.conf:37]: Add a new line?

File: applications/nrf5340_audio/overlay-mcuboot.conf [hunk 7]
```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+CONFIG_PCD_APP=y
+CONFIG_BOARD_ENABLE_CPUNET=n
+
+# Those kconfig should be moved from board file to prj.conf
```
Comment (rick1082) [inline applications/nrf5340_audio/overlay-mcuboot.conf:9]: Shall we remove this note?

File: applications/nrf5340_audio/overlay-mcuboot.conf [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+# We need reduce size of mcuboot
+CONFIG_NFCT_PINS_AS_GPIOS=n
+
+# I2S
+CONFIG_I2S=n # Zephyr I2S driver not used, we are using custom shim
+CONFIG_NRFX_I2S=n
+
+# Enable NRFX_CLOCK for ACLK control
```
Comment (rick1082) [inline applications/nrf5340_audio/overlay-mcuboot.conf:17]: Since all the comments in here describes why the configs need to be enable.
But actually in here we are going to disable all the configs. So we need to update the comments.

**Review by rick1082 — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/overlay-dfu.conf (hunk 4 above)
Comment (pirun) [reply applications/nrf5340_audio/overlay-dfu.conf:25]: Revised as suggested.

File: applications/nrf5340_audio/overlay-dfu.conf (hunk 5 above)
Comment (pirun) [reply applications/nrf5340_audio/overlay-dfu.conf:41]: Revised as suggested.

File: applications/nrf5340_audio/overlay-dfu_external_flash.conf (hunk 6 above)
Comment (pirun) [reply applications/nrf5340_audio/overlay-dfu_external_flash.conf:37]: Revised as suggested.

File: applications/nrf5340_audio/overlay-dfu_external_flash.conf [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+CONFIG_IMG_ERASE_PROGRESSIVELY=y
+CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY=y
+CONFIG_FLASH_NOP_DEVICE=y
+CONFIG_SPI_NOR=y
+CONFIG_FLASH_SHELL=y
+CONFIG_SPI_NOR_CS_WAIT_DELAY=5
+CONFIG_SPI_NOR_SFDP_DEVICETREE=y
+CONFIG_LOG=y
+CONFIG_FLASH_LOG_LEVEL_DBG=y
+# Enable uart driver
```
[the remark this answers is not in the document]
Comment (pirun) [reply applications/nrf5340_audio/overlay-dfu_external_flash.conf:67]: Revised as suggested.

File: applications/nrf5340_audio/overlay-mcuboot.conf (hunk 7 above)
Comment (pirun) [reply applications/nrf5340_audio/overlay-mcuboot.conf:9]: Maybe rephrase here?
I'd like to make a note here which kconfigs should be moved from board config to prj.
When it's time for board config cleanup ,we have reference here.

File: applications/nrf5340_audio/overlay-mcuboot.conf (hunk 8 above)
Comment (pirun) [reply applications/nrf5340_audio/overlay-mcuboot.conf:17]: Update comments from enable to disable.

File: applications/nrf5340_audio/overlay-mcuboot_external_flash.conf [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+CONFIG_FLASH_SIMULATOR_STATS=n
+
+# Enable custom command to erase settings partition.
+CONFIG_ENABLE_MGMT_PERUSER=y
+CONFIG_BOOT_MGMT_CUSTOM_STORAGE_ERASE=y
+CONFIG_HW_CC3XX=n
+CONFIG_NRF_CC3XX_PLATFORM=n
+CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY=y
+CONFIG_BOARD_ENABLE_CPUNET=n
+# Those kconfig should be moved from board file to prj.conf
```
[the remark this answers is not in the document]
Comment (pirun) [reply applications/nrf5340_audio/overlay-mcuboot_external_flash.conf:53]: Maybe rephrase here?
I'd like to make a note here which kconfigs should be moved from board config to prj.
When it's time for board config cleanup ,we have reference here.

**Comment by pirun** [comment]
Rebase to 2022/7/20 main branch commit 05871f9b9c2aebf0a3c188a61b3788baea783180

File: applications/nrf5340_audio/child_image/b0n.conf [hunk 11]
```diff
@@ -0,0 +1,4 @@
+# SPDX-License-Identifier: Apache-2.0
+CONFIG_RPMSG_SERVICE=n
```
Comment (erikrobstad) [inline applications/nrf5340_audio/child_image/b0n.conf:2]: Nitpick: Add a newline between the licence and the configs. This should be done for all .conf files in this PR

File: applications/nrf5340_audio/overlay-dfu.conf [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+CONFIG_BOOTLOADER_MCUBOOT=y
+CONFIG_MCUBOOT_IMAGE_VERSION="0.0.0+0"
+CONFIG_NCS_INCLUDE_RPMSG_CHILD_IMAGE=y
+
+# Increase system workqueue stack size to prevent stack overflows while handling
+# SMP requests.
```
Comment (erikrobstad) [inline applications/nrf5340_audio/overlay-dfu.conf:11]: Nitpick: Remove period. Also on line 35

File: applications/nrf5340_audio/overlay-dfu_external_flash.conf [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+CONFIG_MCUMGR_LOG_LEVEL_DBG=y
+CONFIG_MCUMGR=y
+CONFIG_MCUMGR_CMD_OS_MGMT=y
+CONFIG_MCUMGR_CMD_STAT_MGMT=y
+CONFIG_MCUMGR_CMD_IMG_MGMT=y
+CONFIG_MCUMGR_BUF_SIZE=1024
+CONFIG_MCUBOOT_UTIL_LOG_LEVEL_DBG=y
+
+# Increase system workqueue stack size to prevent stack overflows while handling
+# SMP requests.
```
Comment (erikrobstad) [inline applications/nrf5340_audio/overlay-dfu_external_flash.conf:21]: Nitpick: Remove period. Also on line 38

File: applications/nrf5340_audio/overlay-dfu_external_flash.overlay [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+			30 b0 30 b0  f7 c4 d5 5c  00 be 29 ff  f0 d0 ff ff
+		];
+		size = <67108864>;
+		has-dpd;
+		t-enter-dpd = <10000>;
+		t-exit-dpd = <5000>;
+		dpd-wakeup-sequence = <30000 20 45000>;
+	};
+};
+&gpio_fwd {
```
Comment (erikrobstad) [inline applications/nrf5340_audio/overlay-dfu_external_flash.overlay:87]: Add a newline here

File: applications/nrf5340_audio/prj.conf [hunk 15]
```diff
@@ -8,7 +8,7 @@
 CONFIG_NRF5340_AUDIO=y
 CONFIG_REBOOT=y
 CONFIG_MAIN_THREAD_PRIORITY=10
-CONFIG_MAIN_STACK_SIZE=1940
+CONFIG_MAIN_STACK_SIZE=2800
```
Comment (erikrobstad) [inline applications/nrf5340_audio/prj.conf:11]: What is the reason why we need to increase the stack size?
Now the stack size is larger even if we do not include the DFU functionality, right?

File: applications/nrf5340_audio/shifted_bin/revision.txt [hunk 16]
```diff
@@ -0,0 +1,2 @@
+ble5-ctr-rpmsg_shifted.hex for normal b0n size
```
Comment (erikrobstad) [inline applications/nrf5340_audio/shifted_bin/revision.txt:1]: I do not think we need a .txt file here. This information can be written in the doc (README.rst)

File: applications/nrf5340_audio/src/main.c [hunk 17]
```diff
@@ -207,6 +220,19 @@ void main(void)
 
 	ret = audio_datapath_tone_play(440, 500, 0.2);
 	ERR_CHK(ret);
+	/* Enable MCUMGR */
+	#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
```
Comment (erikrobstad) [inline applications/nrf5340_audio/src/main.c:224]: Nitpick: Swap the places of CONFIG_MCUMGR_CMD_IMG_MGMT and CONFIG_MCUMGR_CMD_OS_MGMT to match the order of #include on line 29-34

File: applications/nrf5340_audio/tools/buildprog/buildprog.py [hunk 18]
```diff
@@ -51,6 +51,16 @@
 COMP_FLAG_RELEASE = NRF5340_AUDIO_FOLDER / "overlay-release.conf"
 COMP_FLAG_DEBUG = NRF5340_AUDIO_FOLDER / "overlay-debug.conf"
 
+#DFU relative overlay
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/buildprog.py:54]: I think we can remove this comment

File: applications/nrf5340_audio/tools/buildprog/buildprog.py [hunk 19]
```diff
@@ -51,6 +51,16 @@
 COMP_FLAG_RELEASE = NRF5340_AUDIO_FOLDER / "overlay-release.conf"
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/buildprog.py:51]: Have the changed code been run through a Python code formatter? I know that the rest of buildprog.py has.

File: applications/nrf5340_audio/tools/buildprog/nrf5340_audio_dk_devices.json [hunk 20]
```diff
@@ -1,6 +1,6 @@
 [
  {
-        "nrf5340_audio_dk_snr": 1000,
+        "nrf5340_audio_dk_snr": 1050161015,
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/nrf5340_audio_dk_devices.json:3]: Remove this change. We do not want to push serial numbers to this file

File: applications/nrf5340_audio/tools/buildprog/fw_info_data.py [hunk 21]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/fw_info_data.py:1]: This file should be run through a Python code formatter

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 19 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py:51]: Like autopep8

File: applications/nrf5340_audio/tools/buildprog/fw_info_data.py (hunk 21 above)
Comment (erikrobstad) [reply applications/nrf5340_audio/tools/buildprog/fw_info_data.py:1]: Like autopep8

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 22]
```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python3
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:1]: This file should be run through a Python code formatter like autopep8

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 23]
```diff
@@ -0,0 +1,273 @@
...
+#
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+"""
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:6]: Newline(s) between license and code

**Comment by erikrobstad** [comment]
I have not looked closely at the Python files.

Some general thoughts:

Several configs are comment out in the different .conf files. There are also several TODOs in the .conf files. These things should not be included in the code that is merged into NCS

There are going to be alot of merge conflicts/problems related to the configs, since our newest code is based on Kconfig.defaults files in stead of .conf files. We must find the best way of solving this

File: applications/nrf5340_audio/child_image/b0n.conf (hunk 11 above)
Comment (pirun) [reply applications/nrf5340_audio/child_image/b0n.conf:7]: Re-format all config file as advised.

File: applications/nrf5340_audio/overlay-dfu_external_flash.conf (hunk 13 above)
Comment (pirun) [reply applications/nrf5340_audio/overlay-dfu_external_flash.conf]: Re-format all config file as advised.

File: applications/nrf5340_audio/overlay-dfu_external_flash.overlay (hunk 14 above)
Comment (pirun) [reply applications/nrf5340_audio/overlay-dfu_external_flash.overlay:88]: Re-format all config file as advised.

File: applications/nrf5340_audio/prj.conf (hunk 15 above)
Comment (pirun) [reply applications/nrf5340_audio/prj.conf]: This might be leftover when debugging sdcard over SPI.
I'll remove it.

File: applications/nrf5340_audio/shifted_bin/revision.txt (hunk 16 above)
Comment (pirun) [reply applications/nrf5340_audio/shifted_bin/revision.txt]: Removed.

File: applications/nrf5340_audio/src/main.c (hunk 17 above)
Comment (pirun) [reply applications/nrf5340_audio/src/main.c]: Re-order function register sequence.

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 19 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py:51]: I have run buildprog.py/fw_info_data.py/pcft_sign.py with  pylint.
_pylint --rcfile=$ZEPHYR_BASE/scripts/ci/pylintrc_
Will run with autopep8 again.

File: applications/nrf5340_audio/tools/buildprog/nrf5340_audio_dk_devices.json (hunk 20 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/nrf5340_audio_dk_devices.json]: SNR accident committed. Will remove.

File: applications/nrf5340_audio/tools/buildprog/buildprog.py (hunk 18 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/buildprog.py]: Removed.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 22 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py]: Will run with autopep8.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 23 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py:7]: Re-format with advised.

**Comment by pirun** [comment]
> I have not looked closely at the Python files.
> 
> Some general thoughts:
> 
> Several configs are comment out in the different .conf files. There are also several TODOs in the .conf files. These things should not be included in the code that is merged into NCS
Remove comment config and TODO message.
> 
> There are going to be alot of merge conflicts/problems related to the configs, since our newest code is based on Kconfig.defaults files in stead of .conf files. We must find the best way of solving this

Maybe we can close this PR.
I can prepare another PR based on BAP_MAIN which is what I am doing.
We can focus on new PR.

File: applications/nrf5340_audio/README.rst [hunk 24]
```diff
@@ -138,7 +138,8 @@ These modules include the following major ones:
 Since the application architecture is uniform and the firmware code is shared, the set of audio modules in use depends on the chosen stream mode (BIS or CIS), the chosen audio inputs and outputs (USB or analog jack), and if the gateway or the headset configuration is selected.
 
 .. note::
-   In the current version of the application, no bootloader is used, and device firmware update (DFU) is not supported.
+   In the current version of the application, bootloader is disabled by default.
+   Device Firmware Update (DFU) can only be enabled when :ref:`nrf53_audio_app_building_script`.
```
Comment (greg-fer) [inline applications/nrf5340_audio/README.rst:142]:
```suggestion
   Device Firmware Update (DFU) can only be enabled when :ref:`nrf53_audio_app_building_script`.
   See :ref:`nrf53_audio_app_configuration_configure_fota` for details.
```

File: applications/nrf5340_audio/README.rst [hunk 25]
```diff
@@ -1334,4 +1389,4 @@ Legal notices for the nRF5340 Audio DK
    To obtain access to the repository, `contact the sales department <Contact Us_>`_.
 
 .. |net_core_hex_note| replace:: The network core for both gateway and headsets is programmed with the precompiled Bluetooth Low Energy Controller binary file :file:`ble5-ctr-rpmsg_<XYZ>.hex`, where ``<XYZ>`` corresponds to the controller version, for example :file:`ble5-ctr-rpmsg_3216.hex`.
-   This file includes the LE Audio Controller Subsystem for nRF53 and is provided in the :file:`applications/nrf5340_audio/bin` directory.
+   This file includes the LE Audio Controller Subsystem for nRF53 and is provided in the :file:`applications/nrf5340_audio/bin` directory. If DFU is enabled, Bluetooth Low Energy Controller binary file will be :file:`pcft_CPUNET.hex` .
```
Comment (greg-fer) [inline applications/nrf5340_audio/README.rst:1392]:
```suggestion
   This file includes the LE Audio Controller Subsystem for nRF53 and is provided in the :file:`applications/nrf5340_audio/bin` directory.
   If :ref:`DFU <nrf53_audio_app_configuration_configure_fota>` is enabled, the subsystem's binary file will be :file:`pcft_CPUNET.hex`.
```

File: applications/nrf5340_audio/README.rst [hunk 26]
```diff
@@ -924,14 +967,26 @@ The building command for running the script requires providing the following par
...
+  .. code-block:: console
 
-.. code-block:: console
+     python buildprog.py -c app -b debug -d both -m internal -M
 
-   python buildprog.py -c app -b debug -d both
+  If you run this command with the ``external`` DFU type parameter instead, the external flash layout will be enabled with the minimal size of the network core bootloader.
 
-This command can be ran from any location, as long as the correct path to :file:`buildprog.py` is given.
+The command can be ran from any location, as long as the correct path to :file:`buildprog.py` is given.
```
Comment (greg-fer) [inline applications/nrf5340_audio/README.rst:989]:
```suggestion
The command can be run from any location, as long as the correct path to :file:`buildprog.py` is given.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
A couple of comments more, to fix typos and tie the different sections more with additional crosslinks.

**Review by greg-fer — APPROVED** [review]

File: applications/nrf5340_audio/README.rst [hunk 27]
```diff
@@ -138,7 +138,9 @@ These modules include the following major ones:
 Since the application architecture is uniform and the firmware code is shared, the set of audio modules in use depends on the chosen stream mode (BIS or CIS), the chosen audio inputs and outputs (USB or analog jack), and if the gateway or the headset configuration is selected.
 
 .. note::
-   In the current version of the application, no bootloader is used, and device firmware update (DFU) is not supported.
+   In the current version of the application, bootloader is disabled by default.
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:141]:
```suggestion
   In the current version of the application, the bootloader is disabled by default.
```

File: applications/nrf5340_audio/README.rst [hunk 28]
```diff
@@ -850,6 +852,46 @@ You can switch to using the I2S serial connection by adding the ``CONFIG_AUDIO_S
 When testing the application, an additional audio jack cable is required to use I2S.
 Use this cable to connect the audio source (PC) to the analog **LINE IN** on the development kit.
 
+.. _nrf53_audio_app_configuration_configure_fota:
+
+Configuring FOTA upgrades
+=========================
+
+You can configure FOTA upgrades to replace the applications on both the application core and the network core.
+The nRF5340 Audio application supports the following types of the DFU flash layouts:
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:861]:
```suggestion
The nRF5340 Audio application supports the following types of DFU flash memory layouts:
```

File: applications/nrf5340_audio/README.rst [hunk 29]
```diff
@@ -850,6 +852,46 @@ You can switch to using the I2S serial connection by adding the ``CONFIG_AUDIO_S
...
+
+* Internal flash layout - which supports only single-image DFU.
+* External flash layout - which supports :ref:`multi-image DFU <ug_nrf5340_multi_image_dfu>`.
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:864]:
```suggestion
* Internal flash memory layout - which supports only single-image DFU.
* External flash memory layout - which supports :ref:`multi-image DFU <ug_nrf5340_multi_image_dfu>`.
```

File: applications/nrf5340_audio/README.rst [hunk 30]
```diff
@@ -850,6 +852,46 @@ You can switch to using the I2S serial connection by adding the ``CONFIG_AUDIO_S
...
+
+LE Audio Controller Subsystem for nRF53 supports both the normal size and the minimal size of bootloader.
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:866]:
```suggestion
The LE Audio Controller Subsystem for nRF53 supports both the normal and minimal sizes of the bootloader.
```

File: applications/nrf5340_audio/README.rst [hunk 31]
```diff
@@ -850,6 +852,46 @@ You can switch to using the I2S serial connection by adding the ``CONFIG_AUDIO_S
...
+The minimal size is specified using the :kconfig:option:`CONFIG_NETBOOT_MIN_PARTITION_SIZE`.
+
+Hardware requirements for external flash DFU
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:869]:
```suggestion
Hardware requirements for external flash memory DFU
```

File: applications/nrf5340_audio/README.rst [hunk 32]
```diff
@@ -850,6 +852,46 @@ You can switch to using the I2S serial connection by adding the ``CONFIG_AUDIO_S
...
+--------------------------------------------
+
+To enable the external flash DFU, you need an additional flash shield.
+The nRF5340 Audio application uses MX25R6435F as the SPI NOR Flash.
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:873]:
```suggestion
To enable the external flash DFU, you need an additional flash memory shield.
The nRF5340 Audio application uses the MX25R6435F as the SPI NOR Flash.
```

File: applications/nrf5340_audio/README.rst [hunk 33]
```diff
@@ -924,14 +968,26 @@ The building command for running the script requires providing the following par
...
+
+See the following examples of the parameter usage with the command run from the :file:`buildprog` directory:
+
+* Example 1: The following command builds the application using the script for the application core with the ``debug`` application version for both the headset and the gateway:
+
+  .. code-block:: console
+
+     python buildprog.py -c app -b debug -d both
+
+* Example 2: The following command builds the application as in the example 1, but with the DFU internal flash layout enabled and the minimal size of the network core bootloader:
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:982]:
```suggestion
* Example 2: The following command builds the application as in *example 1*, but with the DFU internal flash memory layout enabled and using the minimal size of the network core bootloader:
```

File: applications/nrf5340_audio/README.rst [hunk 34]
```diff
@@ -924,14 +968,26 @@ The building command for running the script requires providing the following par
...
 
-For example, to build the application using the script for the application core with the ``debug`` application version for both the headset and the gateway, run the following command from the :file:`buildprog` directory:
+  .. code-block:: console
 
-.. code-block:: console
+     python buildprog.py -c app -b debug -d both -m internal -M
 
-   python buildprog.py -c app -b debug -d both
+  If you run this command with the ``external`` DFU type parameter instead, the external flash layout will be enabled with the minimal size of the network core bootloader.
```
Comment (FrancescoSer) [inline applications/nrf5340_audio/README.rst:988]:
```suggestion
  If you run this command with the ``external`` DFU type parameter instead, the external flash memory layout will be enabled using the minimal size of the network core bootloader.
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]
Just a few suggestions

**Comment by pirun** [comment]
Close. Create new PR https://github.com/nrfconnect/sdk-nrf/pull/8303 for BAP refactoring.
